thirdparty/nginx/nginx-1.7.1-zimbra/src/mail/ngx_mail_proxy_module.c

```diff
@@ -2058,6 +2072,13 @@ static void ngx_mail_proxy_choke_session(ngx_mail_session_t *s)
         ngx_log_debug1(NGX_LOG_DEBUG_MAIL, s->connection->log, 0,
                        "close mail proxy connection: %d",
                        s->proxy->upstream.connection->fd);
+#if (NGX_MAIL_SSL)
+
+        if (s->proxy->upstream.connection->ssl) {
+            s->proxy->upstream.connection->ssl->no_wait_shutdown = 1;
+            ngx_ssl_shutdown(s->proxy->upstream.connection);
+        }
+#endif
 
         ngx_close_connection(s->proxy->upstream.connection);
```

Comment:
I was just wondering that should we be Nullifying the connection just after this statement
Like this:
s->proxy->upstream.connection = NULL;